classes/Layer/Layer.mjs

```diff
@@ -14,53 +17,112 @@ export class Layer {
         this.layer.id = id
     }
 
-    asJSON() {
-        return {
-            type: 'Range',
-            '@context': 'https://iiif.io/api/presentation/3.0/',
-            items: this.layer.body ?? [],
-            target: this.layer.target ?? '',
-        }
-    }
+    async addLayer(layer) {
+        await this.#load()
+        const label = layer?.label ?? `${this.data.label ?? "Default"} - Layer ${Date.now()}`
+        const canvases = layer.canvases
```

Comment:
@thehabes commented on this. The `layer.pages` is correct, not `.canvases`.

Reply:
maybe I'm not following... if you are passing in canvases, the idea is that you are making a new layer by passing in 

```
{ 
    label: "words",
    canvases: [ <URI>, ... ]
}
```

I think where we would be getting these, the Canvas object would be easier/better to pass in.